protocol/contracts/lpp/src/state/config.rs

```diff
@@ -104,9 +103,10 @@ impl Config {
 impl From<InstantiateMsg> for Config {
     fn from(msg: InstantiateMsg) -> Self {
         // 0 is a non-existing code id
```

Comment:
If that really is the case, I feel like `Code` should encapsulate `std::num::NonZeroU64` and `Option<Code>` should be used here. That would provide proper guarantees and there wouldn't be any loss as `Option<NonZeroXYZ>` has the same memory layout as `XYZ` due to special optimizations.

Reply:
this is a logical error, and I'm going to fix it right after this PR is merged. The lease code is going to be provided at contract instantiation.

Reply:
Re: NonZero - no guarantees that it really holds true